lib/index.js

```diff
@@ -45,6 +45,10 @@ internals.schema = Joi.object({
         token: Joi.string().required(),
         headers: Joi.object(),
         profile: Joi.func(),
+        resource: Joi.string().when('name', { is: 'microsoftonline', then: Joi.required(), otherwise: Joi.forbidden() }),
```

Comment:
`otherwise: Joi.optional()` . Someone may create their own custom provider that needs this but not necessarily contribute back to this project.